cumulusci/tasks/salesforce/tests/test_network_member_group.py

```diff
@@ -0,0 +1,594 @@
+
+        # Assert scenario execute as expected.
+        self.assertEqual(
+            f'No {sobject_type} record found with Name "{parent_name}"',
+            context.exception.args[0],
+        )
+
+        task.logger.info.assert_not_called()
+
+        task.logger.warn.assert_not_called()
+
+        task.sf.NetworkMemberGroup.create.assert_not_called()
+
+    def test_create_network_member_group__parent_already_exists(self):
+        network_name = "network_name"
+        sobject_type = "sobject_type"
+        parent_name = "parent_name"
+        parent_id = "parent_id"
+
+        self.assertTrue(parent_id)
+
+        task = create_task(CreateNetworkMemberGroups, {"network_name": network_name})
+
+        task._parent_ids = set()
+        task._parent_ids.add(parent_id)
+        self.assertTrue(parent_id in task._parent_ids)
+
+        task._network_id = "network_id"
+
+        task.logger = Mock()
+        task.logger.info = Mock()
+        task.logger.warn = Mock()
+
+        task.sf = Mock()
+        task.sf.NetworkMemberGroup = Mock()
+
+        insert_response = Mock()
+        insert_response.get = Mock()
+        task.sf.NetworkMemberGroup.create = Mock(insert_response)
+
+        # Execute the test.
+        task._create_network_member_group(sobject_type, parent_name, parent_id)
+
+        # Assert scenario execute as expected.
+        task.logger.info.assert_not_called()
+
+        task.logger.warn.assert_called_once_with(
+            f'        Already exists for "{parent_name}"'
+        )
+
+        task.sf.NetworkMemberGroup.create.assert_not_called()
+
+    def test_create_network_member_group__creating_parent_with_success(self):
+        network_name = "network_name"
+        sobject_type = "sobject_type"
+        parent_name = "parent_name"
+        parent_id = "parent_id"
+
+        self.assertTrue(parent_id)
+
+        task = create_task(CreateNetworkMemberGroups, {"network_name": network_name})
+
+        task._parent_ids = set()
+        self.assertFalse(parent_id in task._parent_ids)
+
+        task._network_id = "network_id"
+
+        task.logger = Mock()
+        task.logger.info = Mock()
+        task.logger.warn = Mock()
+
+        task.sf = Mock()
+        task.sf.NetworkMemberGroup = Mock()
+
+        insert_response = {"success": True}
+        self.assertTrue(insert_response.get("success") is True)
+
+        task.sf.NetworkMemberGroup.create = Mock(return_value=insert_response)
+
+        # Execute the test.
+        task._create_network_member_group(sobject_type, parent_name, parent_id)
+
+        # Assert scenario execute as expected.
+        task.logger.info.assert_called_once_with(f'        "{parent_name}"')
+
+        task.logger.warn.assert_not_called()
+
+        task.sf.NetworkMemberGroup.create.assert_called_once_with(
+            {"NetworkId": task._network_id, "ParentId": parent_id}
+        )
+
+    def test_create_network_member_group__creating_parent_not_success__with_errors(
```

Comment:
Here on down, I'm not going to mark redundant nested mocks or tautological assertions; those notes apply as above.